File: src/frontend/src/types/payment.ts
// Types for Payment
export interface PaymentIntentResponse {
  client_secret: string
  payment_intent_id: string
  amount: number
  currency: string
}

export interface PayNowResponse {
  paynow_qr_url: string
  paynow_reference: string
  expires_at: string
}

File: src/backend/app/schemas/payment.py
# backend/app/schemas/payment.py
from typing import List, Optional

from pydantic import BaseModel


class PaymentIntentCreate(BaseModel):
    order_id: str
    payment_method_types: List[str] = ["card"]


class PaymentIntentResponse(BaseModel):
    client_secret: str
    payment_intent_id: str
    amount: int  # cents
    currency: str = "sgd"


class PayNowConfirmRequest(BaseModel):
    order_id: str
    payment_method_type: str = "paynow"


class PayNowConfirmResponse(BaseModel):
    paynow_qr_url: str
    paynow_reference: str
    expires_at: str


class PaymentStatusResponse(BaseModel):
    status: str  # pending/processing/succeeded/failed
